pkg/local_object_storage/shard/gc.go

```diff
@@ -238,17 +247,54 @@ func (s *Shard) collectExpiredObjects(ctx context.Context, e Event) {
 }
 
 func (s *Shard) collectExpiredTombstones(ctx context.Context, e Event) {
-	expired, err := s.getExpiredObjects(ctx, e.(newEpoch).epoch, func(typ object.Type) bool {
-		return typ == object.TypeTombstone
+	epoch := e.(newEpoch).epoch
+	log := s.log.With(zap.Uint64("epoch", epoch))
+
+	log.Debug("started expired tombstones handling")
+
+	const tssDeleteBatch = 50
+	tss := make([]meta.TombstonedObject, 0, tssDeleteBatch)
+	tssExp := make([]meta.TombstonedObject, 0, tssDeleteBatch)
+
+	iterPrm := new(meta.GraveyardIterationPrm).SetHandler(func(deletedObject meta.TombstonedObject) error {
+		tss = append(tss, deletedObject)
+
+		if len(tss) == tssDeleteBatch {
+			return meta.ErrInterruptIterator
+		}
+
+		return nil
 	})
-	if err != nil || len(expired) == 0 {
+
+	for {
+		log.Debug("iterating tombstones")
```

Comment:
it was added explicitly not for tests

think I could help sometimes: it will be printed every iteration so at least it may say about a number of tombstones